forge-game/src/main/java/forge/game/ability/effects/CopySpellAbilityEffect.java

```diff
@@ -128,32 +130,39 @@ public void resolve(SpellAbility sa) {
                             final Player p = (Player) o;
                             if (p.equals(originalTargetPlayer))
                                 continue;
-                            if (p.isValid(type.split(","), chosenSA.getActivatingPlayer(), chosenSA.getHostCard(), sa)) {
+                            if (p.isValid(type.split(","), sa.getActivatingPlayer(), sa.getHostCard(), sa)) {
```

Comment:
I'm not sure about changing the Valid checks... but without this it misses references to Remembered

I don't think this will break other cards as they are much simpler `CopyForEachCanTarget` stuff